pkg/provisioners/managers/virtualcluster/provisioner.go

```diff
@@ -0,0 +1,300 @@
+	o, _ := options.(*Options)
+
+	return &Provisioner{
+		options: o,
+	}
+}
+
+// Ensure the ManagerProvisioner interface is implemented.
+var _ provisioners.ManagerProvisioner = &Provisioner{}
+
+func (p *Provisioner) Object() unikornv1core.ManagableResourceInterface {
+	return &p.cluster
+}
+
+// getRegionClient returns an authenticated context with a client credentials access token
+// and a client.  The context must be used by subseqent API calls in order to extract
+// the access token.
+func (p *Provisioner) getRegionClient(ctx context.Context, traceName string) (context.Context, regionapi.ClientWithResponsesInterface, error) {
+	cli, err := coreclient.ProvisionerClientFromContext(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return regionutil.Client(ctx, cli, &p.options.clientOptions, p.options.identityOptions, p.options.regionOptions, traceName)
+}
+
+// getKubeconfig reads the kubeconfig from the Kubernetes region.
+func (p *Provisioner) getKubeconfig(ctx context.Context, client regionapi.ClientWithResponsesInterface) ([]byte, error) {
+	organizationID, ok := p.cluster.Labels[coreconstants.OrganizationLabel]
+	if !ok {
+		return nil, fmt.Errorf("%w: cluster missing organization ID", ErrLabel)
+	}
+
+	region, err := regionutil.Region(ctx, client, organizationID, p.cluster.Spec.RegionID)
+	if err != nil {
+		return nil, err
+	}
+
+	return regionutil.Kubeconfig(region)
+}
+
+type remoteKubernetes struct {
+	cluster    *unikornv1.VirtualKubernetesCluster
+	kubeconfig []byte
+}
+
+func (r *remoteKubernetes) ID() *cd.ResourceIdentifier {
+	resourceLabels, _ := r.cluster.ResourceLabels()
+
+	var labels []cd.ResourceIdentifierLabel
+
+	for _, label := range coreconstants.LabelPriorities() {
+		if value, ok := resourceLabels[label]; ok {
+			labels = append(labels, cd.ResourceIdentifierLabel{
+				Name:  label,
+				Value: value,
+			})
+		}
+	}
+
+	return &cd.ResourceIdentifier{
+		Name:   "region-" + r.cluster.Spec.RegionID,
+		Labels: labels,
+	}
+}
+
+func (r *remoteKubernetes) Config(ctx context.Context) (*clientcmdapi.Config, error) {
+	config, err := clientcmd.NewClientConfigFromBytes(r.kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	rawConfig, err := config.RawConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return &rawConfig, nil
+}
+
+func (p *Provisioner) getProvisioner(kubeconfig []byte) provisioners.Provisioner {
+	apps := newApplicationReferenceGetter(&p.cluster)
+
+	remoteGenerator := &remoteKubernetes{
+		cluster:    &p.cluster,
+		kubeconfig: kubeconfig,
+	}
+
+	remoteCluster := remotecluster.New(remoteGenerator, true)
+
+	// TODO: we need some machines to run the cluster on, separate
+	// from the workload pool.  This information and the scheduling
+	// suff needs passing into the provisioner.
+	provisioner := remoteCluster.ProvisionOn(
+		virtualkubernetes.New(apps.vCluster, p.options.domain).InNamespace(p.cluster.Name),
+		// NOTE: If you are using a previsioned physical cluster as a region
```

Comment:
```suggestion
		// NOTE: If you are using a Unikorn-provisioned physical cluster as a region
```
(I _think_ that's what this means)

Reply:
This should just say "pre-provisioned", it doesn't have to be a Unikorn / NKS-provisioned cluster - any that satisfies the Kubernetes pre-reqs basically.